### PR TITLE
Form Validation

### PR DESCRIPTION
**Pull request by steets250**
Checks the sign in and sign up form fields for valid values.

### CODE REVIEW

File: src/components/RegisterForm/index.tsx [hunk 1]
```diff
@@ -42,90 +50,124 @@ const RegisterForm: React.FC<RegisterFormProps> = (props) => {
...
+            <input
+              name="firstName"
+              className="input-box"
+              placeholder="First name"
+              value={values.firstName}
+              onChange={handleChange}
+              onBlur={handleBlur}
+            />
+          </div>
+          <p className="form-error">{errors.firstName ? errors.firstName : null}</p>
```
Comment (leestanley) [inline src/components/RegisterForm/index.tsx:67]: Can you add a color to these error messages - I recommend `#FF6666`

File: src/containers/RegisterForm.tsx [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
+import * as Yup from 'yup';
 
 import RegisterForm from '../components/RegisterForm';
 import { registerAccount } from '../actions/registerActions';
 
+const RegisterSchema = Yup.object().shape({
+  firstName: Yup.string().max(20, 'Too Long').required('Required'),
+  lastName: Yup.string().max(20, 'Too Long').required('Required'),
+  email: Yup.string().email('Invalid Email').required('Required'),
+  password: Yup.string().min(8, 'Too Short').required('Required'),
```
Comment (leestanley) [inline src/containers/RegisterForm.tsx:12]: Min pass word should be 10 not 8 - a 9 length password will throw a random error

File: src/containers/SignInForm.tsx [hunk 3]
```diff
@@ -1,16 +1,25 @@
 import { connect } from 'react-redux';
 import { withFormik } from 'formik';
+import * as Yup from 'yup';
 
 import SignInForm from '../components/SignInForm';
 import { loginUser } from '../actions/authActions';
 
+const SignInSchema = Yup.object().shape({
+  email: Yup.string().email('Invalid Email').required('Required'),
+  password: Yup.string().min(8, 'Too Short').required('Required'),
```
Comment (leestanley) [inline src/containers/SignInForm.tsx:10]: same here

**Review by leestanley — CHANGES_REQUESTED** [review]

**Comment by leestanley** [comment]
image
Input box for major bleeds into year

**Comment by steets250** [comment]
Ok I'll make those changes

DxePkytVAAAb61H

**Comment by leestanley** [comment]
> Ok I'll make those changes
> 
> DxePkytVAAAb61H

File: src/containers/RegisterForm.tsx (hunk 2 above)
Comment (steets250) [reply src/containers/RegisterForm.tsx:12]: Not quite sure what you mean that 9 length gives an error. I just tried it and it worked for sign up and sign in. I'll bump the min up to 9 but I don't think 10 is necessary.

File: src/containers/RegisterForm.tsx [hunk 4]
```diff
@@ -1,9 +1,21 @@
...
+import * as Yup from 'yup';
 
 import RegisterForm from '../components/RegisterForm';
 import { registerAccount } from '../actions/registerActions';
 
+const RegisterSchema = Yup.object().shape({
+  firstName: Yup.string().max(20, 'Too Long').required('Required'),
+  lastName: Yup.string().max(20, 'Too Long').required('Required'),
+  email: Yup.string().email('Invalid Email').required('Required'),
+  password: Yup.string().min(9, 'Too Short').required('Required'),
```
Comment (leestanley) [inline src/containers/RegisterForm.tsx:12]: o lmao its 9 characters
Last nit: can we just specify ('Password must be at least 9 characters') / ('Password must be less than 21 characters')

**Review by leestanley — APPROVED** [review]

**Pull request merged**